src/overlays/actors/ovl_En_Geg/z_en_geg.c

```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x1000, 0x100);
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+        if (sp2E == sp2C) {
+            ActorCutscene_Stop(this->unk_498);
+            this->unk_4AC = 20;
+            func_80BB2020(this, globalCtx);
+            this->actionFunc = func_80BB30B4;
+        } else if (Animation_OnFrame(&this->skelAnime, 24.0f)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOLON_STAND_IMT);
+        }
+    } else if (this->unk_4AC == 4) {
+        if (Animation_OnFrame(&this->skelAnime, 0.0f)) {
+            this->unk_230 |= 1;
+            this->actor.shape.yOffset = 14.0f;
+            if (this->unk_496 == 0xD69) {
+                func_80BB19C0(&this->unk_4E4, this, globalCtx);
+                this->actionFunc = func_80BB2F7C;
+            } else {
+                this->actionFunc = func_80BB3318;
+            }
+        } else if (Animation_OnFrame(&this->skelAnime, 24.0f)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOLON_SIT_IMT);
+        }
+    }
+}
+
+void func_80BB2F7C(EnGeg* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToS(&this->actor.world.rot.y, Math_Vec3f_Yaw(&this->actor.world.pos, &this->unk_4E4), 4, 1000, 1);
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+
+    if ((this->actor.xzDistToPlayer < 150.0f) && (fabsf(this->actor.playerHeightRel) < 10.0f) &&
+        (this->actor.bgCheckFlags & 1)) {
+        this->unk_4AC = 2;
+        this->actor.speedXZ = 0.0f;
+        this->unk_230 &= ~1;
+        this->actor.shape.yOffset = 0.0f;
+        func_80BB2020(this, globalCtx);
+        this->actionFunc = func_80BB2E00;
+    } else {
+        this->actor.speedXZ = 5.0f;
+        Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    }
+
+    if (this->actor.bgCheckFlags & 1) {
+        if (this->unk_230 & 0x80) {
+            func_800B9010(&this->actor, NA_SE_EN_GOLON_SIRLOIN_ROLL - SFX_FLAG);
+        } else {
+            this->unk_230 |= 0x80;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_EYEGOLE_ATTACK);
+        }
+    }
+}
+
+void func_80BB30B4(EnGeg* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        if (player->transformation == PLAYER_FORM_GORON) {
+            this->unk_496 = 0xD6A;
+        } else if (Player_GetMask(globalCtx) == PLAYER_MASK_DON_GEROS_MASK) {
+            this->unk_496 = 0xD89;
+        } else {
+            this->unk_496 = 0xD6E;
+        }
+        func_801518B0(globalCtx, this->unk_496, &this->actor);
+        this->actionFunc = func_80BB27D4;
+        this->actor.flags &= ~0x10000;
+    } else if (this->actor.xzDistToPlayer < 150.0f) {
+        this->actor.flags |= 0x10000;
+        func_800B8614(&this->actor, globalCtx, 150.0f);
+    }
+}
+
+void func_80BB31B8(EnGeg* this, GlobalContext* globalCtx) {
+    s32 sp2C = 0x88;
```

Comment:
```suggestion
    s32 getItemId = GI_MASK_DON_GERO;
```
This is what this is